src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -288,21 +288,23 @@ def instance(userid: Optional[str] = None,
 
         daemon_keys = RgwClient._daemons.keys()
         if not daemon_name:
-            if len(daemon_keys) > 1:
-                try:
-                    multiiste = RgwMultisite()
-                    default_zonegroup = multiiste.get_all_zonegroups_info()['default_zonegroup']
-
-                    # Iterate through _daemons.values() to find the daemon with the
-                    # matching zonegroup_id
-                    for daemon in RgwClient._daemons.values():
-                        if daemon.zonegroup_id == default_zonegroup:
-                            daemon_name = daemon.name
-                            break
-                except Exception:  # pylint: disable=broad-except
-                    daemon_name = next(iter(daemon_keys))
-            else:
-                # Handle the case where there is only one or no key in _daemons
+            try:
+                if len(daemon_keys) > 1:
+                    default_zonegroup = (
+                        RgwMultisite()
+                        .get_all_zonegroups_info()
+                        .get('default_zonegroup')
+                    )
+                    if default_zonegroup:
+                        daemon_name = next(
+                            (daemon.name
+                             for daemon in RgwClient._daemons.values()
+                             if daemon.zonegroup_id == default_zonegroup),
+                            None
+                        )
+                # If daemon_name is still not set, use the first key from daemon_keys
+                daemon_name = daemon_name or next(iter(daemon_keys))
+            except Exception:  # pylint: disable=broad-except
                 daemon_name = next(iter(daemon_keys))
```

Comment:
So what happens if an exception is catched and daemon_keys is `None`?

Reply:
@Pegonzal , daemon_keys can not be None. if any exception is catched, it will continue the normal flow i.e. getting first entry in the daemon_keys

Reply:
It can be any empty dict `{}` in cases where no rgw daemons are found as per the current code.
So we should be mindful of that.

Reply:
@afreen23 , we cannot access rgw through dashboard if we have no rgw daemons, its taken care of in the frontend itself, where we show `No RGW daemons found!` if there's no daemon.

Reply:
What if we call from an API ?

Reply:
Its handled in backend as well, https://github.com/ceph/ceph/blob/368b3cb173a12a58407b418655c270a4f9292bd1/src/pybind/mgr/dashboard/services/rgw_client.py#L85